### PR TITLE
Concurrency logic

### PR DESCRIPTION
**Pull request by kolya182**

### CODE REVIEW

File: test/BlockchainParser.test.ts [hunk 1]
```diff
@@ -22,15 +23,38 @@ describe("Test BlockchainParser", () => {
...
+
+        const tests = [
+            {startBlock: 2, endBlock: 2, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 2, expected: 2},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 10, expected: 2},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 2, expected: 2},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 3, expected: 3},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 4, expected: 3},
```
Comment (vikmeup) [inline test/BlockchainParser.test.ts]: did you try testing ascending and descending logic as well?

File: test/BlockchainParser.test.ts [hunk 2]
```diff
@@ -22,15 +23,38 @@ describe("Test BlockchainParser", () => {
...
+            });
+        });
+    });
+
     describe("Test getNumberBlocks() in forward mode", () => {
         let blockchainParser;
-        const concurrentBlocks = 1;
-        const rebalanceOffsets = [30];
+        const maxConcurrentBlocks = 1;
+        const rebalanceOffsets = [10];
```
Comment (vikmeup) [inline test/BlockchainParser.test.ts]: leave rebalance at 30

File: test/BlockchainParser.test.ts (hunk 1 above)
Comment (kolya182) [reply test/BlockchainParser.test.ts]: Yes

File: test/BlockchainParser.test.ts (hunk 2 above)
Comment (kolya182) [reply test/BlockchainParser.test.ts]: Fixed

**Pull request merged**